### ISSUE TITLE
Pyright raises type error when supplying an attr field with both default and a validator

### ISSUE THREAD

**Issue by bavalpey** [microsoft/pyright#8133]
**Describe the bug**
Pyright incorrectly reports an error when supplying both a validator and a default to `attrs.field`
This does not seem to be a bug with the `attrs` package, as using mypy reports no issues.

**Code or Screenshots**

```python
import attrs
@attrs.define
class Foo:
    a: bool = attrs.field(default=False, validator=attrs.validators.instance_of(bool))
```

![image](https://github.com/microsoft/pyright/assets/36710691/89bfd36d-e162-4a86-a085-da65321dafba)

Interestingly, if the validator is surrounded with parenthesis (which should have no effect), pyright reports no error:

```python
import attrs
@attrs.define
class Foo:
    a: bool = attrs.field(default=False, validator=(attrs.validators.instance_of(bool)))
```

**VS Code extension or command-line**
command-line, version 1.1.367

**Comment by erictraut** [comment]
Thanks for the bug report.

You've actually identified two separate bugs. The first is related to your observation that the behavior changes when you add parens. As you noted, this should have no impact on type evaluation. This bug was easy to address, and the fix will be in the next release.

The second issue is more difficult to fix and is related to the way that pyright's constraint solver is designed. The constraint solver is responsible for "solving" type variables for a generic function call. The bug you've identified here is related to two other open issues: #5855 and #8048. This fix will likely take much longer. In the meantime, if you want to work around it, you can swap the order of the two arguments. Place the `validator` first, then `default`.

**Comment by bavalpey** [comment]
> In the meantime, if you want to work around it, you can swap the order of the two arguments. Place the `validator` first, then `default`.

This doesn't actually fix the error. I was actually fixing it by using the parenthesis. I couldn't figure out a way to fix it other than with an ignore directive.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.374.
